apps/builder/app/canvas/canvas.tsx

```diff
@@ -120,6 +122,29 @@ export const Canvas = ({
   useShortcuts();
   useSharedShortcuts();
 
+  useEffect(() => {
+    const instanceId = data.tree?.root.id;
+    if (instanceId !== undefined) {
+      setDataCollapsed(instanceId);
+    }
+  });
+
+  useEffect(() => {
+    return subscribe("sendStoreChanges", ({ source, changes }) => {
+      for (const change of changes) {
+        if (change.namespace === "styleSourceSelections") {
+          for (const patch of change.patches) {
+            const instanceId = patch.value?.instanceId;
+
+            if (typeof instanceId === "string") {
+              setDataCollapsed(instanceId);
+            }
+          }
+        }
+      }
+    });
+  }, [data.tree?.root.id]);
```

Comment:
- lets move this as a custom hook into collapsed
- separate react from non-react - everything inside useEffect is non-react, make a separate function and call it from the hook

Reply:
This is very hacky assumption and will break next week. Let's move it back to hurt everyone's eyes.

Reply:
Just keep in mind that we need to somehow know what instanceId was affected by the style change.

Reply:
The issue is that token style can affect a few instances. Not just one.